File: 001-020/020.rb
# 基本的には問16の解き方と同じ

# 階乗を求める関数
def factrial(num)
  num.zero? ? 1 : factrial(num - 1) * num
end

result = factrial(100)

puts result.to_s.split('').map(&:to_i).reduce(:+)

File: 011-020/020.js
//通常の整数型では桁あふれを起こすので、BigIntを利用する
//数字の末尾にnを付ける
//BigIntはNode v10の機能
const factrial_n = num => {
  return num === 0n ? 1n : factrial_n(num - 1n) * num;
};

let result = factrial_n(100n);
console.log(result.toString().split('').map(x=>parseInt(x,10)).reduce((a,b)=>a+b));
